frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.unit.spec.tsx

```diff
@@ -141,59 +171,490 @@ describe("AddToDashSelectDashModal", () => {
+        it("should not render public collections", async () => {
+          await setup({
+            card: CARD_IN_PERSONAL_COLLECTION,
+            collections: COLLECTIONS,
+          });
+
+          expect(
+            screen.getByRole("heading", {
+              name: PERSONAL_COLLECTION.name,
+            }),
+          ).toBeInTheDocument();
+          expect(
+            screen.queryByRole("heading", {
+              name: COLLECTION.name,
+            }),
+          ).not.toBeInTheDocument();
+        });
+
+        describe('"Create a new dashboard" option', () => {
+          it('should not render "Create a new dashboard" option when opening the root collection (public collection)', async () => {
+            await setup({
+              card: CARD_IN_PERSONAL_COLLECTION,
+            });
+
+            expect(
+              screen.queryByRole("heading", {
+                name: "Create a new dashboard",
+              }),
+            ).not.toBeInTheDocument();
+          });
+
+          it('should render "Create a new dashboard" option when opening personal subcollections', async () => {
+            await setup({
+              card: CARD_IN_PERSONAL_COLLECTION,
+            });
+
+            userEvent.click(
+              screen.getByRole("heading", {
+                name: PERSONAL_COLLECTION.name,
+              }),
+            );
+
+            expect(
+              screen.getByRole("heading", {
+                name: "Create a new dashboard",
+              }),
+            ).toBeInTheDocument();
+
+            userEvent.click(
+              screen.getByRole("heading", {
+                name: PERSONAL_SUBCOLLECTION.name,
+              }),
+            );
+
+            expect(
+              screen.getByRole("heading", {
+                name: "Create a new dashboard",
+              }),
+            ).toBeInTheDocument();
+          });
+        });
+
+        describe("whether we should render dashboards in a collection", () => {
+          it("should not render dashboards when opening the root collection (public collection)", async () => {
+            const dashboardInPublicCollection = createMockDashboard({
+              id: 5,
+              name: "Dashboard in public collection",
+              // `null` means it's in the root collection
+              collection_id: null,
+              model: "dashboard",
+            });
+
+            await setup({
+              card: CARD_IN_PERSONAL_COLLECTION,
+              dashboard: dashboardInPublicCollection,
+            });
+
+            await waitFor(() => {
+              expect(screen.queryByText("Loading...")).not.toBeInTheDocument();
+            });
+
+            expect(
+              screen.queryByRole("heading", {
+                name: dashboardInPublicCollection.name,
+              }),
+            ).not.toBeInTheDocument();
+          });
+
+          it("should render dashboards when opening personal collections", async () => {
```

Comment:
test Query builder > Newly created question > #6 from [the epic](https://github.com/metabase/metabase/issues/34733)